packages/react-components/src/localization/LocalizationProvider.tsx

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { createContext, useState, useMemo, useContext, useCallback, useEffect } from 'react';
+import { defaultLocaleStringsLoader, locales } from './loadLocaleStrings';
+import defaultStrings from './translated/en-US.json';
+
+const LOCALE_CACHE_KEY = 'AzureCommunicationUI_Locale';
+
+export interface ILocale {
+  locale: string;
+  englishName: string;
+  displayName: string;
+  rtl: boolean;
+}
+
+/**
+ * Collection of ILocale
+ */
+export type ILocaleCollection = Record<string, ILocale>;
+
+export type ILocaleKeys = Record<string, string>;
+
+export interface ILocaleContext {
+  locale: ILocale;
+  locales: ILocaleCollection;
+  strings: ILocaleKeys;
+  setLocale: (locale: string, forceReload?: boolean) => void;
+}
+
+const defaultLocaleContext: ILocaleContext = {
+  locale: locales['en-US'],
+  locales: locales,
+  strings: defaultStrings,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
+  setLocale: (locale: string, forceReload?: boolean) => {}
+};
+
+export const LocaleContext = createContext<ILocaleContext>(defaultLocaleContext);
+
+export type LocalizationProviderProps = {
+  initialLocale: string;
+  locales: ILocaleCollection;
+  localeStringsLoader?: (locale: string) => Promise<ILocaleKeys>;
+  storage?: Storage;
+  children: React.ReactNode;
+};
+
+export const LocalizationProvider = (props: LocalizationProviderProps): JSX.Element => {
+  const { children, initialLocale, locales, storage, localeStringsLoader } = props;
+
+  useEffect(() => {
+    if (storage) {
+      const cachedLocale = storage.getItem(LOCALE_CACHE_KEY);
```

Comment:
I don't think we want to require the use of local storage at the component layer. 
This should be an implementation detail that contoso needs to figure out

We should say if you give us a big set of strings we'll disperse them across the composite for you